pybossa/api/__init__.py

```diff
@@ -707,7 +707,11 @@ def get_service_request(task_id, service_name, major_version, minor_version):
         if isinstance(service, dict):
             url = '{}/{}/{}/{}'.format(proxy_service_config['uri'], service_name, major_version, minor_version)
             headers = service.get('headers')
-            ret = requests.post(url, headers=headers, json=payload['data'])
+            ssl_cert = current_app.config.get('SSL_CERT_PATH')
+            if ssl_cert:
+                ret = requests.post(url, headers=headers, json=payload['data'], verify=ssl_cert)
+            else:
+                ret = requests.post(url, headers=headers, json=payload['data'])
```

Comment:
Please confirm if `verify=None` would work for http conn. If this works, the code can be simplified as below w/o if/else block
```
ssl_cert = current_app.config.get('SSL_CERT_PATH')
ret = requests.post(url, headers=headers, json=payload['data'], verify=ssl_cert)
```

Reply:
No, but I can re-write it more succinctly.

Looking [at the docs](https://requests.readthedocs.io/en/latest/api/#requests.post:~:text=of%20the%20proxy.-,verify%20%E2%80%93%20(optional)%20Either%20a%20boolean%2C%20in%20which%20case%20it%20controls%20whether%20we%20verify%20the%20server%E2%80%99s%20TLS%20certificate%2C%20or%20a%20string%2C%20in%20which%20case%20it%20must%20be%20a%20path%20to%20a%20CA%20bundle%20to%20use.%20Defaults%20to%20True.,-stream%20%E2%80%93%20(optional)%20if), the valid values are either Boolean or string.

> verify – (optional) Either a boolean, in which case it controls whether we verify the server’s TLS certificate, or a string, in which case it must be a path to a CA bundle to use.

Reply:
@dchhabda I've update the code, lmk what you think.